src/jit/registerxmm.h

```diff
@@ -8,6 +8,10 @@
 #error  Must define REGDEF macro before including this file
 #endif
 
+#ifndef LEGACY_BACKEND
+#error This file is only used for the LEGACY_BACKEND build.
+#endif
+
```

Comment:
Since we have retired amd64 legacy backend, we can get rid of target_amd64 specific xmm reg definitions from this file?

In fact we can delete this file altogether by moving legacy x86 xmm reg definitions to register.h file.